Development/src/dataProvider.js

```diff
@@ -371,28 +399,50 @@ async function convertHTTPResponseToDataProvider(
                 }
 
                 let connectionAddresses = {};
-                let connectionAddress;
-                if (deviceJSONData.hasOwnProperty('controls')) {
-                    for (let i in deviceJSONData.controls)
-                        connectionAddresses[
-                            deviceJSONData.controls[i]['type']
-                        ] = deviceJSONData.controls[i]['href'];
-                } else {
+                deviceJSONData.controls.forEach(function(control) {
+                    const type = control.type.replace('.', '_');
+                    if (type.startsWith('urn:x-nmos:control:sr-ctrl')) {
+                        if (!connectionAddresses.hasOwnProperty(type)) {
+                            set(connectionAddresses, type, [control.href]);
+                        } else {
+                            connectionAddresses[type].push(control.href);
+                        }
+                    }
+                });
+                // Return IS-04 if no Connection API endpoints
+                if (
+                    Object.keys(connectionAddresses).length === 0 &&
+                    connectionAddresses.constructor === Object
```

Comment:
Minor: I guess you copied this condition from somewhere like https://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object, but the rationale given for adding the `.constructor` check isn't applicable here.